src/Agent.Worker/Release/Artifacts/FileContainer/ContainerFetchEngine.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Services.Agent.Worker;
+using Microsoft.VisualStudio.Services.FileContainer;
+
+namespace Agent.Worker.Release.Artifacts.FileContainer
+{
+    public class ContainerFetchEngine : FetchEngine
+    {
+        public ContainerFetchEngine(
+            IExecutionContext logger,
+            Uri collectionUrl,
+            string projectName,
+            long containerId,
+            string rootItemPath,
+            string rootDestinationDir,
+            int parallelDownloadLimit,
+            bool verbose,
+            string accessToken)
+            : base(logger, collectionUrl, projectName, containerId, rootItemPath, rootDestinationDir, parallelDownloadLimit, verbose, accessToken)
+        {
+        }
+
+        public async Task FetchAsync()
+        {
+            List<FileContainerItem> containerItems = await ItemCache.GetItemsAsync().ConfigureAwait(false);
```

Comment:
Are we missing something in the rest of the project? Does your concern apply here, or is it a concern that only applies to other types of apps (e.g. WPF or IIS)?